Sources/FuturedArchitecture/Architecture/Coordinator.swift

```diff
@@ -46,10 +47,12 @@ public extension Coordinator {
 public protocol TabCoordinator: Coordinator {
     associatedtype Tab: Hashable
 
+    @MainActor
     var selectedTab: Tab { get set }
 }
 
 public protocol NavigationStackCoordinator: Coordinator {
+    @MainActor
```

Comment:
Is it necessary to add `@MainActor` wrappers to those variables ? 🤔 

We wanna that developers will use the dedicated functions (navigate, pop, ...) for setting those variables. 
Those functions are already on MainActor. Ofc if they will be setting those variable directly, the runtime warnings will appear but I may be some hint that they are doing something worng. 🤔

Reply:
We provide those properties as a protocol requirement. Therefore (by definition), those variables are accessible to the "user." The extensions do not cover _all_ possible cases of flow modification (for example, you might want to "push" or "pop" several scenes at once.

The API provided by the extensions do _promote_ but not _ensure_ safety. We should not expose an API which we know is prone to _runtime_ errors, especially if it's avoidable. 

I have even considered removing the extension altogether, but decided to keep them as an API sugar for most common cases.

Reply:
As I have alluded to at earlier occasion, I don't think we should limit ourselves to only using `protocol`s. If there is a variable and we _know_ such variable is **required** and also poses a runtime risk, we should hide such property. In such case, we should use `class`.